qt/applications/workbench/workbench/plotting/figureinteraction.py

```diff
@@ -811,19 +811,20 @@ def _change_plot_normalization(self, ax):
             if arg_set['workspaces'] in ax.tracked_workspaces:
                 workspace = ads.retrieve(arg_set['workspaces'])
                 arg_set['distribution'] = is_normalized
-                arg_set_copy = copy(arg_set)
-                [
-                    arg_set_copy.pop(key)
-                    for key in ['function', 'workspaces', 'autoscale_on_update', 'norm']
-                    if key in arg_set_copy.keys()
-                ]
                 if 'specNum' not in arg_set:
                     if 'wkspIndex' in arg_set:
                         arg_set['specNum'] = workspace.getSpectrum(
                             arg_set.pop('wkspIndex')).getSpectrumNo()
                     else:
                         raise RuntimeError("No spectrum number associated with plot of "
                                            "workspace '{}'".format(workspace.name()))
+
+                arg_set_copy = copy(arg_set)
+                [
+                    arg_set_copy.pop(key)
+                    for key in ['function', 'workspaces', 'autoscale_on_update', 'norm']
+                    if key in arg_set_copy.keys()
+                ]
```

Comment:
```suggestion
                arg_set_copy = copy(arg_set)
                for key in ['function', 'workspaces', 'autoscale_on_update', 'norm']:
                    try:
                        del arg_set_copy[key]
                    except KeyError:
                        continue
```

I see how moving this part fixes the issue, well found. Since this part of code is touched, I would propose to improve it in the suggested way: there seems to be no reason to use list comprehension to create a list that is then unassigned to anything and thus lost, as well as using `pop` instead of explicit deletion, as the value related to the removed key is not captured nor needed.

Reply:
Yes I agree this is much clearer, and I suspect faster.